### PR TITLE
Beta with modern aiormq

### PR DESCRIPTION
**Pull request by mosquito**

### CODE REVIEW

**Review by decaz — CHANGES_REQUESTED** [review]

File: aio_pika/channel.py [hunk 1]
```diff
@@ -59,12 +59,13 @@ def __init__(
...
         self._done_callbacks = CallbackCollection(self)
         self._return_callbacks = CallbackCollection(self)
-        self._channel = None  # type: Optional[aiormq.Channel]
+        self._channel: Optional[aiormq.Channel] = None
         self._channel_number = channel_number
         self._on_return_raises = on_return_raises
         self._publisher_confirms = publisher_confirms
 
         self._delivery_tag = 0
+        self._closed: bool = False
```
[the remark this answers is not in the document]
Comment (mosquito) [reply aio_pika/channel.py:68]: That's indicates the channel manually closed by user

File: aio_pika/message.py [hunk 2]
```diff
@@ -180,23 +189,20 @@ def header_converter(value: Any) -> bytes:
     ).encode()
 
 
-@header_converter.register(bytes)
```
[the remark this answers is not in the document]
Comment (mosquito) [reply aio_pika/message.py:183]: because bytes is the default now

File: aio_pika/queue.py [hunk 3]
```diff
@@ -42,23 +41,22 @@ def __init__(
...
         self.arguments = arguments
         self.passive = passive
-        self.declaration_result = None  # type: aiormq.spec.Queue.DeclareOk
+        self.declaration_result: Optional[aiormq.spec.Queue.DeclareOk] = None
         self._get_lock = asyncio.Lock()
 
     @property
-    def channel(self) -> aiormq.Channel:
-        if self._channel is None:
+    def __channel(self) -> aiormq.Channel:
```
Comment (decaz) [inline aio_pika/queue.py:56]: What do you think about naming it `_channel`?

**Review by decaz — CHANGES_REQUESTED** [review]

File: setup.py [hunk 4]
```diff
@@ -29,16 +29,16 @@
...
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: Implementation :: PyPy",
         "Programming Language :: Python :: Implementation :: CPython",
     ],
     packages=find_packages(exclude=["tests"]),
     package_data={"aio_pika": ["py.typed"]},
-    install_requires=["aiormq>=3.2.3,<4", "yarl"],
+    install_requires=["aiormq~=5.1.0", "yarl"],
```
[the remark this answers is not in the document]
Comment (mosquito) [reply setup.py]: WIP

**Comment by julianfortune** [comment]
Upgrading to this branch seems to have resolved an issue my team was having (similar to #288) and also allows setting the `timeout` parameter via URI query params; I know you are busy, so if you need any assistance with development or testing please let me know! Thank you for all of your work on this ❤️

**Comment by mosquito** [comment]
@julianfortune Thank you for these kind words. Of course, I need help with testing, since there are a lot of changes and I tried to make the old tests pass, but there are still a lot of changes and testing will be required. If you find a bug, fell free to write comments in this review.

**Comment by mosquito** [comment]
Rebased and reworked in #436

**Pull request closed without merging**